Automattic/wp-calypso
# BBE : WebsiteContent : Add video support for the video gallery

## Proposed Changes

* Provides video support for DIFM uploads
* Refactors existing image upload code to support `media` uploads which currently includes any images or videos and is expandable.

<img width="1094" alt="image" src="https://user-images.githubusercontent.com/3422709/202625038-7c1feb42-1d3a-49b4-8523-a6b3d99724ca.png">

#### Video Size Limits Not Handled
This PR or BBE in general has not considered the possibility of exceeding the plan's storage limit. Currently the premium plans allow 13 GB. With images we allow at the worst case 48 images and it would have never realistically exceeded this size. But with videos this can quite easily be exceeded. This needs to be handled and will be covered in a subsequent PR.

Another edge case can be when a user uploads several high capacity videos. And then remove them and try to add the same or other videos. Here a major concern is that we do not "cleanup" media when they are removed from the UI. This will also have to be addressed now that users might upload high capacity videos.



#### Testing Instructions

* Complete a purchase `start/do-it-for-me` with a video gallery page
* Try uploading videos makes sure the video icon and file name is visible as shown above
* Upload images to the log section and contact section to make sure the image upload has not regressed.
* Upload images to any other section to make sure everything works
* Submit content and make sure all the image and video urls are available on the site build post.
  
#### Pre-merge Checklist





Fixes Automattic/martech#1234
